pkg/controller/apiservicecabundle/apiservice_cabundle_controller.go

```diff
@@ -0,0 +1,158 @@
+package apiservicecabundle
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	kapierrors "k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	apiregistrationapiv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	apiserviceclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1"
+	apiserviceinformer "k8s.io/kube-aggregator/pkg/client/informers/externalversions/apiregistration/v1"
+	apiservicelister "k8s.io/kube-aggregator/pkg/client/listers/apiregistration/v1"
+)
+
+const (
+	InjectCABundleAnnotationName = "service.alpha.openshift.io/inject-cabundle"
+)
+
+type ServiceServingCertUpdateController struct {
+	apiServiceClient apiserviceclient.APIServicesGetter
+
+	// Services that need to be checked
+	queue workqueue.RateLimitingInterface
+
+	apiServiceLister    apiservicelister.APIServiceLister
+	apiServiceHasSynced cache.InformerSynced
+
+	caBundle []byte
+
+	// syncHandler does the work. It's factored out for unit testing
+	syncHandler func(serviceKey string) error
+}
+
+func NewAPIServiceCABundleInjector(apiServiceInformer apiserviceinformer.APIServiceInformer, apiServiceClient apiserviceclient.APIServicesGetter, caBundle []byte) *ServiceServingCertUpdateController {
+	sc := &ServiceServingCertUpdateController{
+		apiServiceClient: apiServiceClient,
+
+		queue: workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+
+		caBundle: caBundle,
+	}
+
+	sc.apiServiceLister = apiServiceInformer.Lister()
+	apiServiceInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    sc.addAPIService,
+			UpdateFunc: sc.updateAPIService,
+		},
+	)
+	sc.apiServiceHasSynced = apiServiceInformer.Informer().GetController().HasSynced
+
+	sc.syncHandler = sc.syncAPIService
+
+	return sc
+}
+
+func (c *ServiceServingCertUpdateController) syncAPIService(key string) error {
+	_, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	apiService, err := c.apiServiceLister.Get(name)
+	if kapierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	if strings.ToLower(apiService.Annotations[InjectCABundleAnnotationName]) != "true" {
+		return nil
+	}
+	if reflect.DeepEqual(apiService.Spec.CABundle, c.caBundle) {
```

Comment:
@deads2k this is comparing two []byte fields?

Reply:
> @deads2k this is comparing two []byte fields?

Yeah.  If they don't match, we need to update.